src/gnu_debuglink.rs

```diff
@@ -0,0 +1,80 @@
+use reader::{Reader, ReaderOffset};
+use endianity::{EndianBuf, Endianity};
+use parser::Result;
+use Section;
+
+/// The `GnuDebuglink` struct represents the external debuginfo information
+/// found in the `.gnu_debuglink` section.
+#[derive(Debug, Clone, Copy)]
+pub struct GnuDebuglink<R: Reader> {
+    gnu_debuglink_section: R,
+}
+
+impl<'input, Endian> GnuDebuglink<EndianBuf<'input, Endian>>
+where
+    Endian: Endianity
+{
+    /// Constructs a new `GnuDebuglink` instance from the data in the
+    /// `.gnu_debuglink` section.
+    ///
+    /// It is the caller's responsibility to read the `.gnu_debuglink` section
+    /// and present it as a `&[u8]` slice. This means using some ELF loader on
+    /// Linux, a Mach-O loader on OSX, etc.
+    ///
+    /// ```
+    /// use gimli::{GnuDebuglink, LittleEndian};
+    ///
+    /// # let buf = [0x00, 0x01, 0x02, 0x03];
+    /// # let read_gnu_debuglink_section_somehow = || &buf;
+    /// let gnu_debuglink = GnuDebuglink::new(read_gnu_debuglink_section_somehow(), LittleEndian);
+    /// ```
+    pub fn new(gnu_debuglink_section: &'input [u8], endian: Endian) -> Self {
+        Self::from(EndianBuf::new(gnu_debuglink_section, endian))
+    }
+}
+
+impl<R: Reader> GnuDebuglink<R> {
+    /// Returns the filename of the debug information file.
+    pub fn filename(&self) -> Result<R> {
+        self.gnu_debuglink_section.clone().read_null_terminated_slice()
+    }
+
+    /// Returns the CRC32 checksum of the debug information file.
+    pub fn crc32(&self) -> Result<u32> {
+        let mut section = self.gnu_debuglink_section.clone();
+        let base = section.len() - ReaderOffset::from_u32(4);
```

Comment:
This works fine for valid data, but doesn't handle invalid data very well. It'd be better to have a single method that reads both the filename and crc32. I guess it doesn't matter too much though because the crc won't match if it's wrong.

Reply:
Yeah, I wasn't sure what the right API is here - is there another section type that's similar? It seems like the others currently supported are all sequences.

Reply:
I didn't see any other sections that are similar. I think just a `fn read(&self) -> Result<(R, u32)>` would be fine.